src/main/python/ttconv/imsc/elements.py

```diff
@@ -89,6 +89,14 @@ def process_space_attribute(self, parent_ctx: TTMLElement.ParsingContext, xml_el
       space_attr_value = imsc_attr.XMLSpaceAttribute.extract(xml_elem)
       self.space = space_attr_value if space_attr_value is not None else parent_ctx.space
 
+  class WritingContext:
+    '''State information when writing a TTML element'''
+
+    def __init__(self, frame_rate: imsc_attr.TemporalAttributeWritingContext):
```

Comment:
`frame_rate` should be a `Fraction` instance, isn't it?

Reply:
Good catch. Yes it should.